projects/vdk-core/plugins/vdk-plugin-control-cli/src/vdk/plugin/control_cli_plugin/vdk_plugin_control_cli.py

```diff
@@ -41,3 +43,9 @@ def vdk_command_line(root_command: click.Group):
     default_options = DefaultOptions(plugins)
     if default_options.get_default_map():
         root_command.context_settings["default_map"] = default_options.get_default_map()
+
+
+@hookimpl(tryfirst=True)
+def vdk_configure(config_builder: ConfigurationBuilder) -> None:
+    """ """
+    control_service_configuration.add_definitions(config_builder)
```

Comment:
This is totally unrelated to this change, but seeing this code pattern makes me feel weird. Usually in a OOP paradigm, if we execute `A.method(B)` we expect A to mutate, and its mutation to depend on B in some way; however, here we see the opposite, where `config_builder` mutates depending on the `control_service_configuration` object. I don't know if this is something that requires refactoring necessarily or if I'm being pedantic (or maybe I'm entirely wrong and this isn't that unusual, who knows).

Reply:
I am open to better suggestions. You may have a point. We need to be consistent though. 

Another thing I don't like I did was using get_xxx() methods in the DomainOrientedConfiguration class. Python properties would have been a more pythonic way to do it. get_xxx is very java like. But I am sticking to it everywhere until we decide to change it everywhere.